### ISSUE TITLE
training

### ISSUE THREAD

**Issue by zw-92** [verlab/accelerated_features#48]
Hello, I want to do some fine-tuning based on your trained model and my own dataset. Since there is no label, I want to replace the coco20k dataset with my own dataset, or add some of my own dataset to it. Do you recommend "full training" or "training with existing weights"? Do I need to modify the training hyperparameters and strategies? Looking forward to your reply.

**Comment by guipotje** [comment]
Hello @zw-92, thank you for the interest in our work!

I believe you should test the two approaches, but I first recommend a fine-tuning just to see if XFeat improves on your image distribution. This should be faster, use less data, and you can get some experience in training the model. Then, you could just do a from-scratch training, as the model is quite small and you don't need high-end hardware. Regarding hyperparameters, I think the defaults are a good start, then you can test some variations based on the training dynamics.

**Comment by zw-92** [comment]
Thank you for your reply. I trained on MegaDepth and coco20k using the default parameters, but the results, when tested in the same scene with the weights you provided, show a difference of about 300 in the average number of matched feature points—one being 1500 and the other 1200. Do you know why this might be? I also fine-tuned using the original weights and default parameters, but the results were mediocre. After reducing the learning rate, the average number of matches began to approach the results of the original weights, occasionally even surpassing them by about 100 points. However, I lack ground truth data, making it difficult to assess. Do you have any suggestions?

**Comment by 1303463822** [comment]
Hello, I want to fine-tune using open-source weights on my own dataset. How should I set up my model initialization and parameters?
